### ISSUE TITLE
Renovate constantly closes and reopens PR for submodule updates

### ISSUE THREAD

**Issue by okainov** [renovatebot/renovate#10634]
**How are you running Renovate?**

- [ ] WhiteSource Renovate hosted app on github.com
- [x] Self hosted

If using the hosted app, please skip to the next section.
Otherwise, if self-hosted, please complete the following:

Please select which platform you are using:

- [x] github.com

Renovate version: latest Docker

**Describe the bug**

Renovate constantly keeps closing and re-opening PRs for submodule updates:

![image](https://user-images.githubusercontent.com/918446/123601169-a7f9ec00-d7f7-11eb-8cab-cf9ebcefc93d.png)

...

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
[2021-06-28T04:38:18.186Z] DEBUG: remainingBranches=renovate/lib-external-rapidjson-digest (repository=myorg/myrepo)
[2021-06-28T04:38:18.186Z] DEBUG: findPr(renovate/lib-external-rapidjson-digest, undefined, open) (repository=myorg/myrepo)
[2021-06-28T04:38:18.186Z] DEBUG: Found PR #90 (repository=myorg/myrepo)
[2021-06-28T04:38:18.186Z]  INFO: Autoclosing PR (repository=myorg/myrepo)
[2021-06-28T04:38:18.186Z]        "branchName": "renovate/lib-external-rapidjson-digest",
[2021-06-28T04:38:18.186Z]        "prNo": 90,
[2021-06-28T04:38:18.186Z]        "prTitle": "chore(deps): update lib/external/rapidjson commit hash to b557259"
[2021-06-28T04:38:18.186Z] DEBUG: updatePr(90, chore(deps): update lib/external/rapidjson commit hash to b557259 - autoclosed, body) (repository=myorg/myrepo)
[2021-06-28T04:38:19.131Z] DEBUG: PR updated (repository=myorg/myrepo)
[2021-06-28T04:38:19.131Z]        "pr": 90
[2021-06-28T04:38:21.689Z] DEBUG: Deleted remote branch (repository=myorg/myrepo)
[2021-06-28T04:38:21.689Z]        "branchName": "renovate/lib-external-rapidjson-digest"
[2021-06-28T04:38:21.689Z] DEBUG: No local branch to delete (repository=myorg/myrepo)
[2021-06-28T04:38:21.689Z]        "branchName": "renovate/lib-external-rapidjson-digest"
[2021-06-28T04:38:21.689Z] DEBUG: Repository timing splits (milliseconds) (repository=myorg/myrepo)
[2021-06-28T04:38:21.689Z]        "splits": {"init": 15032, "extract": 3154, "lookup": 1117, "update": 12586},
[2021-06-28T04:38:21.689Z]        "total": 36478```
```
</details>

It would be great and useful for debugging if Renovate would print the reason of autoclosure in the logs and not just "autoclosing"...

**Comment by okainov** [comment]
@rarkins any details what I can get from the logs to help you investigate it? I can try to create a test repo using Github App, but I have no confidence it behaves the same way, so I would rely on our repos...

**Comment by rarkins** [comment]
The log you gave originally is not helpful - it's at the end where Renovate closes any branch that it thinks shouldn't exist. It has no knowledge of why the branch existed in the first place, just that it shouldn't exist now.

**Comment by okainov** [comment]
>Does it happen EVERY second run, or less reliably?

> If it's reliable, you should be able to search for the dependency name and branch name it affects and compare what it says on runs which open the PR and ones which autoclose.

As you see from screenshot, like 100+ cycles to close-open. When I was looking for the log, I scrolled down to 3-4 jobs I think (we execute Renovate every hour). So I'd say not every second run, but something more like every 4-6th run, but still quite reliably...

Which part of the logs you mean, which name of the branch? The job which has closed PR (from the log in the my post in the beginning) has `"renovate/lib-external-rapidjson-digest"`

In the previous job (the one which run before the closing one):
```
[2021-06-28T03:37:44.071Z] DEBUG: Found existing PR (repository=myorg/myrepo, branch=renovate/lib-external-rapidjson-digest)
[2021-06-28T03:37:44.071Z] DEBUG: Processing existing PR (repository=myorg/myrepo, branch=renovate/lib-external-rapidjson-digest)
[2021-06-28T03:37:44.071Z] DEBUG: Pull Request #90 does not need updating (repository=myorg/myrepo, branch=renovate/lib-external-rapidjson-digest)
```
Or
```
      "origin/renovate/lib-external-rapidjson-digest": {
[2021-06-28T04:38:08.105Z]              "current": false,
[2021-06-28T04:38:08.105Z]              "name": "origin/renovate/lib-external-rapidjson-digest",
[2021-06-28T04:38:08.105Z]              "commit": "c5c73cd",
[2021-06-28T04:38:08.105Z]              "label": "chore(deps): update lib/external/rapidjson commit hash to b557259"
```

>just that it shouldn't exist now.

This part I think I understand. But I do not understand, *why* Renovate thinks so... What I also can see is that in the list of `DEBUG: packageFiles with updates` I also see no new commit for this given submodule (while there were some update for another submodule). So it looks like it's not able(?) to check commits for this repo for some random reason? But I don't understand what can I do here...

**Comment by rarkins** [comment]
I suggest you create a new repo with only that submodule as the only dependency and try to reproduce the problem there. If so then it will be much easier for you to look through the logs. It's also pretty important that you identify the pattern of when the PR is opened and when it's autoclosed.

**Comment by okainov** [comment]
I tried to create public repo as reproducer, but I think GithubApp-Renovate is not running every time (but on some hooks/changes?). So I'm not sure how can I reproduce the behavior we have internally when we really run Renovate every ~hour.

https://github.com/okainov/renovate-sandbox/pull/1

I also tried closing/renaming the PR to let Renovate create a new one - and the behavior was still the same...

**Comment by rarkins** [comment]
@okainov if you check app.renovatebot.com you can see that Renovate is in fact running regularly on the repo. Therefore it's a case of this repo not reproducing.

**Comment by okainov** [comment]
The issue is still there though

**Comment by viceice** [comment]
> This bug report has been closed as we need a reproduction to work on this. If the original poster or anybody else with the same problem discovers that they can reproduce it, please create a new issue, and reference this issue.

@okainov  ☝️

**Comment by okainov** [comment]
@viceice it's quite obvious that it's not efficient way. It is well reproducible for our team and I tried my best to provide you as much info as I could. But if renovate just doesn't have enough info in the logs, what else can I do? And you just ignored it and closed the issue, so what's the point?

**Comment by viceice** [comment]
> Hi there,
> 
> Help us by making a minimal reproduction repository.
> 
> Before we can start work on your issue we first need to know exactly what's causing the current behavior. A minimal reproduction helps us with this.
> 
> To get started, please read our guide on creating a [minimal reproduction](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to understand what is needed.
> 
> We may close the issue if you (or someone else) have not provided a minimal reproduction within two weeks. If you need more time, or are stuck, please ask for help or more time in a comment.
> 
> Good luck,
> 
> The Renovate team

@okainov ☝️  👉 [minimal reproduction](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md)

**Comment by okainov** [comment]
@viceice  if you would at least pretend you have read comments above, you would notice that we're using private org and I did try to create similar repo with public renovate, but it didn't work. So there is not much more I can do apart of asking you to improve logging.
